src/psyclone/psyGen.py

```diff
@@ -2433,6 +2431,39 @@ def gen_code(self, parent):
             for call in reprod_red_call_list:
                 call.reduction_sum_loop(parent)
 
+    def begin_string(self):
+        '''Returns the beginning statement of this directive, i.e.
+        "omp parallel". The visitor is responsible for adding the
+        correct directive begin.
+
+        :returns: the beginning statement oforthis directive.
```

Comment:
maybe s/beginning/opening/ and s/oforthis/of this/.

Reply:
Done.

Reply:
Thanks but you've got "$!" instead of "!$".

Reply:
Oops - we really need to support compilation tests.